py4j-python/src/py4j/java_gateway.py

```diff
@@ -210,6 +210,12 @@ def find_jar_path():
     #   the jar file is here: virtualenvpath/share/py4j/py4j.jar
     paths.append(os.path.join(os.path.dirname(
             os.path.realpath(__file__)), "../../../../share/py4j/" + jar_file))
+    # pip install py4j # On Windows without admin privileges
```

Comment:
Hm .. is this version specific difference? I have Windows env but it has always worked fine.

Reply:
I am not sure. I only know that I have to fix the path on my machine (my colleagues have the same issue). You can find the version numbers in the linked issue.

Reply:
Let's probably add an if-else based on OS. I would like to minimize the impact when other OSes are used.